joblib/test/test_numpy_pickle.py

```diff
@@ -178,28 +195,48 @@ def test_numpy_persistence():
             for item in obj_:
                 nose.tools.assert_true(isinstance(item, np.ndarray))
             # And finally, check that all the values are equal.
-            nose.tools.assert_true(np.all(np.array(obj) ==
-                                                np.array(obj_)))
+            np.testing.assert_array_equal(np.array(obj), np.array(obj_))
 
         # Now test with array subclasses
-        for obj in (
-                    np.matrix(np.zeros(10)),
-                    np.core.multiarray._reconstruct(np.memmap, (), np.float)
-                   ):
+        for obj in (np.matrix(np.zeros(10)),
+                    np.core.multiarray._reconstruct(np.memmap, (), np.float)):
             this_filename = filename + str(random.randint(0, 1000))
             filenames = numpy_pickle.dump(obj, this_filename,
-                                          compress=compress,
-                                          cache_size=cache_size)
+                                          compress=compress)
+            # All is cached in one file
+            nose.tools.assert_equal(len(filenames), 1)
+
             obj_ = numpy_pickle.load(this_filename)
-            if (type(obj) is not np.memmap
-                        and hasattr(obj, '__array_prepare__')):
+            if (type(obj) is not np.memmap and
+                    hasattr(obj, '__array_prepare__')):
                 # We don't reconstruct memmaps
                 nose.tools.assert_true(isinstance(obj_, type(obj)))
+                np.testing.assert_array_equal(obj_, obj)
```

Comment:
Should you not be testing even when obj is a np.memmap? I.e. could this line could be dedented by 4 spaces?